ibm_cloud_sdk_core/utils.py

```diff
@@ -235,17 +227,22 @@ def __read_from_vcap_services(service_name: str) -> dict:
                 if vcap_service_credentials and isinstance(vcap_service_credentials, dict):
                     break
                 if services[key][i].get('name') == service_name:
-                    vcap_service_credentials = services[key][i].get('credentials')
+                    vcap_service_credentials = services[key][i].get('credentials', {})
         if not vcap_service_credentials:
             if service_name in services.keys():
-                vcap_service_credentials = services.get(service_name)[0].get('credentials')
+                service = services.get(service_name)
+                if service:
+                    vcap_service_credentials = service[0].get('credentials', {})
 
         if vcap_service_credentials and isinstance(vcap_service_credentials, dict):
+            new_vcap_creds = {}
             if vcap_service_credentials.get('username') and vcap_service_credentials.get('password'): # cf
-                vcap_service_credentials['AUTH_TYPE'] = 'basic'
-                vcap_service_credentials['USERNAME'] = vcap_service_credentials.get('username')
-                vcap_service_credentials['PASSWORD'] = vcap_service_credentials.get('password')
+                new_vcap_creds['AUTH_TYPE'] = 'basic'
+                new_vcap_creds['USERNAME'] = vcap_service_credentials.get('username')
+                new_vcap_creds['PASSWORD'] = vcap_service_credentials.get('password')
+                vcap_service_credentials = new_vcap_creds
             elif vcap_service_credentials.get('apikey'):  # rc
-                vcap_service_credentials['AUTH_TYPE'] = 'iam'
-                vcap_service_credentials['APIKEY'] = vcap_service_credentials.get('apikey')
+                new_vcap_creds['AUTH_TYPE'] = 'iam'
+                new_vcap_creds['APIKEY'] = vcap_service_credentials.get('apikey')
```

Comment:
One minor detail... if the VCAP_SERVICES entry that we're copying values from ALSO contains an "iam_apikey" field, then we want to use that instead of the "apikey" field.   So "iam_apikey" trumps "apikey".   Whichever one we find, it should be copied into the new_vcap_creds['APIKEY'] entry in the dictionary.
If you look at the "v0.5.2" version of base_service.py in the python core repo you can see the logic that used to be there in terms of copying values from the vcap services entry.